test/e2e/framework/kcp.go

```diff
@@ -271,16 +272,26 @@ func (c *kcpServer) loadCfg() error {
 			loadError = fmt.Errorf("failed to read admin kubeconfig after kcp start: %w", err)
 			return
 		}
+
 		rawConfig, err := clientcmd.LoadFromFile(kubeconfigPath)
 		if err != nil {
 			loadError = fmt.Errorf("failed to load admin kubeconfig: %w", err)
 			return
 		}
+		// LoadFromFile can return an empty default config if it read an empty file, so
```

Comment:
On L266-274 we have an explicit wait with a `os.Stat` - could we instead use the file info there to wait until the file exists *and* it has a non-zero size?

Reply:
I considered that after I wrote it this way. Would you prefer that?

Reply:
I think it would be cleaner as to your intent and would not require us to assume the behavior of LoadFromFile / be resilient to changes there